client/components/Assignments/AssignmentFilterPanel.tsx

```diff
@@ -0,0 +1,140 @@
+});
+
+class AssignmentFilterPanelComponent extends React.Component<IProps, IState> {
+    constructor(props) {
+        super(props);
+        this.onChangeHandler = this.onChangeHandler.bind(this);
+        this.onChangeMultiple = this.onChangeMultiple.bind(this);
+        this.onClear = this.onClear.bind(this);
+        this.search = this.search.bind(this);
+        this.state = {params: cloneDeep(this.props.currentParams || {})};
+    }
+
+    onClear() {
+        this.setState({params: {}});
+        this.props.setSearchParams({});
+    }
+
+    onChangeHandler<T extends keyof IAssignmentSearchParams>(field: T, value: IAssignmentSearchParams[T]) {
+        const params = cloneDeep(this.state.params);
+
+        if (Array.isArray(value) && value.length === 0) {
+            set(params, field, null);
+        } else {
+            set(params, field, value);
+        }
+
+        this.setState({params: params});
+    }
+
+    onChangeMultiple(updates: IAssignmentSearchParams) {
+        const params = cloneDeep(this.state.params);
+
+        Object.keys(updates).forEach((field) => {
+            const value = get(updates, field);
+
+            if (Array.isArray(value) && value.length === 0) {
+                set(params, field, null);
+            } else {
+                set(params, field, value);
+            }
+        });
+
+        this.setState({params: params});
+    }
+
+    search() {
+        this.props.setSearchParams(this.state.params);
+    }
+
+    render() {
+        const {gettext} = superdeskApi.localization;
+
+        return (
+            <SidePanel
+                shadowLeft={true}
+                testId="search-panel"
+            >
+                <Header className="side-panel__header--border-b">
+                    <Tools
+                        tools={[{
+                            icon: 'close-small',
+                            onClick: this.props.toggleFilterPanel,
+                            title: gettext('Close'),
+                        }]}
+                    />
+                    <h3 className="side-panel__heading">
+                        {gettext('Advanced filters')}
+                    </h3>
+                </Header>
+                <Content>
+                    <ContentBlock>
+                        <ContentBlockInner>
+                            <AdvancedSearch
+                                params={this.state.params}
+                                onChange={this.onChangeHandler}
+                                onChangeMultiple={this.onChangeMultiple}
+                                searchProfile={this.props.searchProfile}
+                                enabledField="enabled"
+                                filterUsersByDesk={this.props.currentDeskId}
+                            />
+                        </ContentBlockInner>
+                    </ContentBlock>
+                </Content>
+                <Footer className="side-panel__footer--button-box">
```

Comment:
convert `Footer` prop-types to typescript intefaces - it is currently showing a compilation error `Type 'Element' is not assignable to type 'ReactNodeLike'.`